author/topics/document-format/text.adoc

```diff
@@ -325,6 +325,55 @@ by the caption.
 Word's description of "`Alt Text`" is:
 "`How would you describe this object and its context to someone who is blind?`".
 
+== Source code
+
+=== Pseudocode
+
+Source code is marked up as elsewhere in Asciidoc, as a preformatted source snippet to
+be rendered in monospace font, and with spaces preserved:
+
+[source,asciidoc]
+--
+[source,ruby]
+----
+  def increment(x)
+    x + 1
+  end
+----
+--
+
+Source code highlighting can be used automatically to highlight keywords specific
+to the nominated computer language.
+
+In several standards, pseudocode is required to be entered. Unlike sourcecode,
+pseudocode is typically in a proportional font, but it still needs to be indented
+to reflect code structure. Moreover, pseudocode typically requires source code highlighting
+such as boldface; but unlike well-defined computer languages, there is no guaranteed way
+of automating such highlighting.
+
+Pseudocode is supported in Metanorma as a special class of example, marked up
+with a pseudocode block macro. Text within a pseudocode block is treated as normal
+text, including respect for inline formatting; lines need to separated by line breaks
+(`+`). However, indentation spaces at the start of each line are preserved, by converting
+them into non-breaking spaces. Initial tabs are converted into four non-breaking spaces.
+
+
+[source,asciidoc]
+--
+[pseudocode]
+====
+*do in-parallel* +
+  [smallcap]#SharedAccess# +
+*enddo*
+
+[smallcap]#ExclusiveAccess# ≡ +
```

Comment:
@opoudjis could you also use some math in the example? For example, `∀` should really be `stem:[forall]`.